drivers/ph_oem/ph_oem.c

```diff
@@ -0,0 +1,446 @@
+                          gpio_mode, gpio_flank, cb, arg) < 0) {
+
+            DEBUG("\n[ph_oem debug] Initilizing enable gpio pin failed.\n");
+            return PH_OEM_GPIO_INIT_ERR;
+        }
+    }
+    return PH_OEM_OK;
+}
+
+int ph_oem_set_interrupt_pin(const ph_oem_t *dev, ph_oem_irq_option_t option)
+{
+    i2c_acquire(I2C);
+
+    if (i2c_write_reg(I2C, ADDR, PH_OEM_REG_INTERRUPT, option, 0x0) < 0) {
+        DEBUG("\n[ph_oem debug] Setting interrupt pin to option %d failed.\n", option);
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_set_led_state(const ph_oem_t *dev, ph_oem_led_state_t state)
+{
+    i2c_acquire(I2C);
+
+    if (i2c_write_reg(I2C, ADDR, PH_OEM_REG_LED, state, 0x0) < 0) {
+        DEBUG("\n[ph_oem debug] Setting LED state to %d failed.\n", state);
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_enable_device(const ph_oem_t *dev, bool enable)
+{
+    if (dev->params.enable_pin == GPIO_UNDEF) {
+        return PH_OEM_ENABLE_GPIO_UNDEF;
+    }
+
+    if (enable) {
+        gpio_set(EN_PIN);
+    }
+    else {
+        gpio_clear(EN_PIN);
+    }
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_set_device_state(const ph_oem_t *dev, ph_oem_device_state_t state)
+{
+    i2c_acquire(I2C);
+
+    if (i2c_write_reg(I2C, ADDR, PH_OEM_REG_HIBERNATE, state, 0x0) < 0) {
+        DEBUG("\n[ph_oem debug] Setting device state to %d failed\n", state);
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+/* polling PH_OEM_REG_NEW_READING register as long as it does not equal 0x01 */
+static int ph_oem_new_reading_available(const ph_oem_t *dev)
+{
+    int8_t new_reading_available;
+
+    i2c_acquire(I2C);
+    do {
+        if (i2c_read_reg(I2C, ADDR, PH_OEM_REG_NEW_READING,
+                         &new_reading_available, 0x0) < 0) {
+            DEBUG("\n[ph_oem debug] Failed at reading PH_OEM_REG_NEW_READING\n");
+            i2c_release(I2C);
+            return PH_OEM_READ_ERR;
+        }
+        xtimer_usleep(20 * US_PER_MS);
+    } while (new_reading_available == 0);
+
+    /* need to manually reset register back to 0x00 */
+    if (i2c_write_reg(I2C, ADDR, PH_OEM_REG_NEW_READING, 0x00, 0x0) < 0) {
+        DEBUG("\n[ph_oem debug] Resetting PH_OEM_REG_NEW_READING failed\n");
+        i2c_release(I2C);
+        return PH_OEM_WRITE_ERR;
+    }
+    i2c_release(I2C);
+
+    return PH_OEM_OK;
+}
+
+int ph_oem_start_new_reading(const ph_oem_t *dev)
+{
+    if (ph_oem_set_device_state(dev, PH_OEM_TAKE_READINGS) < 0) {
+        return PH_OEM_WRITE_ERR;
+    }
+
+    /* if interrupt pin undefined, poll till new reading was taken */
+    if (dev->params.interrupt_pin == GPIO_UNDEF) {
+        if (ph_oem_new_reading_available(dev) < 0) {
+            return PH_OEM_READ_WRITE_ERR;
```

Comment:
Why introduce a new error code just for this case when the actual error reason (either Read or Write) can be known?

Reply:
I removed this error code and added the the actual error reason